### PR TITLE
login: XGI Z7/Z9 (XG20 core) graphic chip requires master-of-seat to …

### PR DESCRIPTION
**Pull request by fbuihuu**
…be set

This graphic chip doesn't have a DRM driver and fallback to vesa-framebuffer
driver.

Without this patch, users of such chip suddenly see their GUI broken without
any indication or reason of what happened (no error message). Hence this
regression is near to impossible to troubleshoot for end users.

### CODE REVIEW

**Comment by hramrach** [comment]
Note: fallout of #10435

**Review by bluca — APPROVED** [review]
Superficially looks fine, but will wait for another pair of eyes.

**Comment by yuwata** [comment]
Superficially LGTM too. But maybe, in future, we should provide a better way to handle such exceptional devices with hwdb or so. But, yeah, that should be out of scope of this PR.

**Comment by poettering** [comment]
I am sorry, this looks wrong to me. First of all this adds the "master-of-seat" tag, without the "seat" tag. So this can't even work correctly.

Conceptually I am not convinced that encoding whether something has a driver or not in udev rules is really the right thing to do. Isn't there a better place to encode this in? 

Did anyone actually request support for this and test this? is there a bug report for this? Adding new support for old obsolete hw sound iffy to me. udev rules are executed linearly, we should have less of thse not more.

Moreover we should stop adding entries like this that just match vid/pid. Do that in hwdb. Thre's already one entry exactly like this,in the rules files, so this makes two, and two is a bunch and should get their own hwdb file really.

**Comment by hramrach** [comment]
Isn't "seat" added by another rule already?

Yes, people actually complained, filed a bug, and tested this rule. https://bugzilla.opensuse.org/show_bug.cgi?id=1187154

This is not for obsolete hardware. This is for current-day virtualized hardware.

Adding the rule does not stop you from consolidating the rules into a hwdb in a separate cleanup. Not having hwdb for these rules should not be an excuse for breaking user's systems.

**Comment by poettering** [comment]
> Isn't "seat" added by another rule already?

i don't see any.

**Comment by poettering** [comment]
> Yes, people actually complained, filed a bug, and tested this rule. https://bugzilla.opensuse.org/show_bug.cgi?id=1187154
> 
> This is not for obsolete hardware. This is for current-day virtualized hardware.

The commit message and commend in the added udev rule suggests otherwise. The bugzilla link was not provided. Please subit patches with full and correct context.

**Comment by poettering** [comment]
Which virtualizer is this supposed to be? the bugzilla report desn't mention that afaics?

**Comment by poettering** [comment]
> Adding the rule does not stop you from consolidating the rules into a hwdb in a separate cleanup. Not having hwdb for these rules should not be an excuse for breaking user's systems.

Sorry, not how this works. We don't merge stuff that maybe hides an issue that noone noticed for 3 years. We fix things properly.

Wouldn't it make more sense if you X11 package that carries support for drm-less graphics would carry a rule for this?

**Comment by fbuihuu** [comment]
> > Isn't "seat" added by another rule already?
> 
> i don't see any.

`SUBSYSTEM=="graphics", KERNEL=="fb[0-9]*", TAG+="seat"` ?

**Comment by fbuihuu** [comment]
> The commit message and commend in the added udev rule suggests otherwise. The bugzilla link was not provided. Please subit patches with full and correct context.

Sorry I forgot to provide the bugzilla link in the commit message, will do next times.

**Comment by fbuihuu** [comment]
> Sorry, not how this works. We don't merge stuff that maybe hides an issue that noone noticed for 3 years. We fix things properly.

Well Leap is not following upstream version closely (there're too many releases anyway and it's hard to make things stable in this context). But  the issue was reported as soon as Leap was upgraded to version containing the commit that decided to drop support for framebuffers.

And I don't think that was an official decision to drop support for framebuffer devices otherwise I would have expected to described in NEWS as it's a backward incompatible change.

**Comment by poettering** [comment]
> `SUBSYSTEM=="graphics", KERNEL=="fb[0-9]*", TAG+="seat"` ?

that's on the fb device, not the pci device? i.e. one level down.

**Comment by fbuihuu** [comment]
> > `SUBSYSTEM=="graphics", KERNEL=="fb[0-9]*", TAG+="seat"` ?
> 
> that's on the fb device, not the pci device? i.e. one level down.

Indeed, but the rule has been tested and was reported to work...

**Comment by hramrach** [comment]
> Which virtualizer is this supposed to be? the bugzilla report desn't mention that afaics?

HyperV and possibly virtualbox as well. It is mentioned in the bug report.

**Comment by keszybz** [comment]
@fbuihuu I applies @poettering's revert. This is a regression, but not a recent one, so we don't need to block the release for it. Please submit a new patch that adds the entry to hwdb.

File: src/login/71-seat.rules.in
Comment (keszybz) [inline src/login/71-seat.rules.in:33]: Also "fallback" → "falls back".

**Comment by fbuihuu** [comment]
> @fbuihuu I applies @poettering's revert. This is a regression, but not a recent one, so we don't need to block the release for it.

As I already tried to explain the regression is only seen now because a bunch of users haven't updated systemd yet. But I don't think we said that this should block the release, at least I haven't but the milestone label.

> Please submit a new patch that adds the entry to hwdb.

OK I'll rework the patch to use hwdb.

**Comment by fbuihuu** [comment]
> OK I'll rework the patch to use hwdb.

PTAL at https://github.com/systemd/systemd/pull/20092

**Pull request merged**
